src/sql/engine/expr/ob_expr_convert_tz.cpp

```diff
@@ -0,0 +1,267 @@
+    input3.set_calc_collation_type(session->get_nls_collation());
+  }
+  return ret;
+}
+
+int ObExprConvertTZ::calc_result3(common::ObObj &result,
+                                  const common::ObObj &input1,
+                                  const common::ObObj &input2,
+                                  const common::ObObj &input3,
+                                  common::ObExprCtx &expr_ctx) const
+{
+  int ret = OB_SUCCESS;
+  if (OB_UNLIKELY(input1.is_null() || input2.is_null() || input3.is_null())) {
+    result.set_null();
+  } else {
+    int64_t timestamp_data = input1.get_timestamp();
+    if (OB_FAIL(calc_convert_tz(timestamp_data, input2.get_string(), input3.get_string(), 
+                                  expr_ctx.my_session_))) {
+      LOG_WARN("convert time zone failed", K(ret));
+    } else {
+      result.set_timestamp(timestamp_data);
+    }
+  }
+  return ret;
+}                                  
+
+
+int ObExprConvertTZ::find_time_zone_pos(const ObString &tz_name,
+                                        const ObTimeZoneInfo &tz_info,
+                                        ObTimeZoneInfoPos *&tz_info_pos)
+{
+  int ret = OB_SUCCESS;
+  tz_info_pos = NULL;
+  ObTZInfoMap *tz_info_map = NULL;
+  if (OB_ISNULL(tz_info_map = const_cast<ObTZInfoMap *>(tz_info.get_tz_info_map()))) {
+    ret = OB_ERR_UNEXPECTED;
+    LOG_WARN("tz_info_map is NULL", K(ret));
+  } else if (OB_FAIL(tz_info_map->get_tz_info_by_name(tz_name, tz_info_pos))) {
+    LOG_WARN("fail to get_tz_info_by_name", K(tz_name), K(ret));
+    tz_info_map->id_map_.revert(tz_info_pos);
+    tz_info_pos = NULL;
+  } else {
+    tz_info_pos->set_error_on_overlap_time(tz_info.is_error_on_overlap_time());
+    // need to test overlap time
+  }
+  return ret;
+}
+
+int ObExprConvertTZ::parse_string(int64_t &timestamp_data, const ObString &tz_str, ObSQLSessionInfo *session, int32_t &offset)
+{
+	int ret = OB_SUCCESS;
+	int ret_more = 0;
+	if (OB_FAIL(ObTimeConverter::str_to_offset(tz_str, offset, ret_more,
+                              false /* oracle_mode */, true /* need_check_valid */)))
+	{
+		LOG_INFO("direct str_to_offset failed");
+		if (OB_LIKELY(OB_ERR_UNKNOWN_TIME_ZONE == ret)){
+			const ObTimeZoneInfo *tz_info = NULL;
+      ObTimeZoneInfoPos *target_tz_pos = NULL;
+      if (OB_ISNULL(tz_info = TZ_INFO(session))) {
+        ret = OB_ERR_UNEXPECTED;
+        LOG_WARN("tz info is null", K(ret), K(session));
+      } else if (OB_FAIL(find_time_zone_pos(tz_str, *tz_info, target_tz_pos))){
+        LOG_WARN("find time zone position failed", K(ret), K(ret_more));
+        if (OB_ERR_UNKNOWN_TIME_ZONE == ret && OB_SUCCESS != ret_more) {
+          ret = ret_more;
+        }
+      } else if (OB_FAIL(calc(timestamp_data, *target_tz_pos, offset))) {
+        LOG_WARN("calc failed", K(ret), K(timestamp_data));
+      } 
+		} else {
+      LOG_WARN("str to offset failed", K(ret));
+    }
+	} else {
+			LOG_INFO("str to offset succeed", K(tz_str), K(offset));
```

Comment:
remove or use LOG_DEBUG.
log that prints for each row should not be info level.